modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-web/src/main/resources/META-INF/resources/js/AdminTooltip.js

```diff
@@ -0,0 +1,245 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+import ClayAutocomplete from '@clayui/autocomplete';
+import ClayButton from '@clayui/button';
+import ClayCard from '@clayui/card';
+import {useResource} from '@clayui/data-provider';
+import ClayDropDown from '@clayui/drop-down';
+import ClayForm, {ClayInput, ClayRadio, ClayRadioGroup} from '@clayui/form';
+import PropTypes from 'prop-types';
+import React, {useLayoutEffect, useState} from 'react';
+
+const AdminTooltip = ({
+	namespace,
+	setRemovePinHandler,
+	setShowTooltip,
+	showTooltip,
+	updatePin,
+}) => {
+	const [pinLabel, setPinLabel] = useState(showTooltip.details.label);
+	const [linkedValue, setLinkedValue] = useState(
+		showTooltip.details.linked_to_sku
+	);
+	const [sku, setSku] = useState(showTooltip.details.sku);
+	const [quantity, setQuantity] = useState(showTooltip.details.quantity);
+	const [networkStatus, setNetworkStatus] = useState(4);
+	const initialLoading = networkStatus === 1;
+	const loadingAd = networkStatus < 4;
+	const error = networkStatus === 5;
```

Comment:
I would put these codes as constants at the top of the file and reference the variable.

For example

```js
const NETWORK_STATUS_MAP = {
	initialLoading: 1,
	error: 5
};

// ...

const initialLoading = networkStatus === NETWORK_STATUS_MAP.initialLoading;
const error = networkStatus === NETWORK_STATUS_MAP.error;
```

Reply:
hello @bryceosterhaus we changed all of this stuff in the [next pr](https://github.com/liferay-frontend/liferay-portal/pull/1374) so we doesn't neet this code anymore anyway thank you

Reply:
@garofalog can you close this PR since the new PR will supersede this one?

Reply:
It's a problem because the new one is incremental. In this case we'll lose this changes in this pr